src/routes/embedded/auth/auth-email-signup/auth-email-verify.view.tsx

```diff
@@ -61,159 +66,150 @@ export function AuthEmailVerifyEmbeddedView() {
       });
 
-      if (error) {
-        toast.error(error.message || "Invalid verification code");
-        return;
-      }
-
       toast.success("Email verified successfully");
-      await new Promise((resolve) => {
-        const interval = setInterval(() => {
-          if (authStatusRef.current === "noWallets") {
-            resolve(null);
-            clearInterval(interval);
-          }
-        }, 100);
-      });
-      navigate(EmbeddedPaths.WalletHomeEmbeddedView);
     } catch (error) {
-      toast.error("Error verifying email");
-    } finally {
       setIsVerifying(false);
+      toast.error(getFriendlyAuthErrorMessage(error, "Invalid verification code"));
     }
+
+    // We leave isVerifying = true intentionally as the user will simply be redirected after verifying the account.
   };
 
-  const verifyOtp = useCallback(() => {
-    if (!email) return;
+  const focusInput = useCallback((inputOrIndex: HTMLInputElement | number) => {
+    const input =
+      typeof inputOrIndex === "number"
+        ? inputRefs.current[Math.min(Math.max(inputOrIndex, 0), OTP_LENGTH - 1)]
+        : inputOrIndex;
 
-    const otpCode = otpDigits.join("");
-    if (otpCode.length !== OTP_LENGTH) {
-      toast.error(`Please enter all ${OTP_LENGTH} digits of the verification code`);
-      return;
-    }
+    if (!input) return;
+
+    input.focus();
 
-    handleVerifyCode(otpCode);
-  }, [email, otpDigits]);
+    requestAnimationFrame(() => {
+      input.select();
+    });
+  }, []);
 
-  const handlePaste = useCallback(
-    (e: React.ClipboardEvent<HTMLInputElement>, index: number) => {
+  const handleInputOrPaste = useCallback(
+    (e: React.ClipboardEvent<HTMLInputElement> | React.ChangeEvent<HTMLInputElement>) => {
       e.preventDefault();
-      const pastedData = e.clipboardData.getData("text/plain").trim();
-      const sanitizedValue = pastedData.replace(/[^0-9]/g, "");
 
-      if (sanitizedValue) {
-        const newDigits = [...otpDigits];
+      const inputs = inputRefs.current;
+      const rawValue =
+        "clipboardData" in e ? e.clipboardData.getData("text/plain").trim() : e.currentTarget.value.trim();
+      const sanitizedValue = rawValue.replace(/[^0-9]/g, "");
+      const currentIndex = parseInt(e.currentTarget.name.replace("otp-input-", "")) || 0;
 
-        // Fill in digits starting from current position
-        for (let i = 0; i < sanitizedValue.length && index + i < OTP_LENGTH; i++) {
-          newDigits[index + i] = sanitizedValue[i];
-        }
+      if (!inputs) return;
 
-        // Find next empty input or focus last input
-        const nextEmptyIndex = newDigits.findIndex((digit, idx) => idx >= index && !digit);
-        const focusIndex = nextEmptyIndex !== -1 ? nextEmptyIndex : OTP_LENGTH - 1;
-        focusInput(focusIndex);
+      let nextEmptyIndex = currentIndex;
 
-        setOtpDigits(newDigits);
-        autoSubmitIfComplete(newDigits);
+      if (sanitizedValue.length === OTP_LENGTH) {
```

Comment:
If pasting a whole code, take it all, ignoring in what `input` the paste took place.